client/src/actions/index.js

```diff
@@ -103,17 +112,47 @@ export const deleteStream = id => async dispatch => {
     });
 };
 
-export const fetchProfile = id => async dispatch => {
+const createProfile = createData => async dispatch => {
+  const payload = {
+    bio: "This is a default bio",
+    email: createData.email,
+    name: createData.name,
+    userid: createData.id,
+    username: createData.name
+  };
   database
     .collection("users")
-    .where("userid", "==", id)
+    .doc(createData.id)
+    .set(payload)
+    .then(() => {
+      dispatch({ type: CREATE_PROFILE, payload: payload });
+      dispatch({ type: SIGN_IN });
+      history.push("/");
+  })
+  .catch(function(error) {
+    console.error("Error cannot create profile: " + error);
+    dispatch({ type: AUTH_FAIL });
+  });
+};
+
+export const fetchProfile = fetchData => async dispatch => {
+  database
+    .collection("users")
+    .where("userid", "==", fetchData.id)
     .get()
     .then(querySnapshot => {
-      const data = querySnapshot.docs.map(doc => doc.data());
-      dispatch({ type: FETCH_PROFILE, payload: data[0] });
+      if(querySnapshot.empty){
+        dispatch(createProfile(fetchData));
```

Comment:
Should this be dispatched here? I'm not too sure how this works or if it even matters, but createProfile is also dispatching some things.

Reply:
Yes this should. It basically calls the createProfile function but since its asynchronous I need to dispatch it.  The createProfile function dispatches the actual action to create the profile. 

This is just for cosmetic purposes to separate the code for createProfile from fetchProfile. Functionality wise it works the same.